File: 94.二叉树的中序遍历.py
#
# @lc app=leetcode.cn id=94 lang=python3
#
# [94] 二叉树的中序遍历
#

# @lc code=start
# Definition for a binary tree node.
from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

# 中序遍历
# 左 根 右
class Solution:

    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        r = []

        def inorder(root):
            if root:
                inorder(root.left)
                r.append(root.val)
                inorder(root.right)

        inorder(root)
        return r


s = Solution()
r = s.inorderTraversal()
print(r)
# @lc code=end

File: 94.二叉树的中序遍历.js
/*
 * @lc app=leetcode.cn id=94 lang=javascript
 *
 * [94] 二叉树的中序遍历
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function (root) {
  let result = []
  var helper = function (root) {
    if (root) {
      helper(root.left)
      result.push(root.val)
      helper(root.right)
    }
  }
  helper(root)
  return result
};
console.table(["apples", "oranges", "bananas"]);
console.time("reticulating splines");
console.timeLog("reticulating splines");
console.table({ a: 1 })
// @lc code=end

